### PR TITLE
Small changes for GitLab

### PR DESCRIPTION
**Pull request by JackDunnNZ**
I have been experimenting with getting docs to build on GitLab and found a few things to fix up.

For reference and to help anyone else, I've been using the following hacked together script  that calls `git_push` to skip all of the travis-specific processing in `deploydocs`:

```julia
if haskey(ENV, "CI")
  tag = get(ENV, "CI_COMMIT_TAG", "")
  tag_ok = isempty(tag) || occursin(Base.VERSION_REGEX, tag)

  if tag_ok
    root = pwd()
    build_dir = "docs/build"
    target_repo = "gitlab.com/xxx/yyy"

    mktempdir() do temp
      devurl = "dev"
      Documenter.git_push(root, temp, target_repo;
          target=build_dir,
          tag=tag,
          host="gitlab.com",
          key=ENV["DOCUMENTER_KEY"],
          # Defaults from Documenter.deploydocs
          sha=readchomp(`git rev-parse --short HEAD`),
          devurl=devurl,
          versions=["stable" => "v^", "v#.#", devurl => devurl],
          forcepush=false,
      )
    end
  end
end
```

There are three commits each with a minor change:

1. Add a `host` option to `git_push` so that the url rewriting and temp ssh config file work for gitlab.com urls as well
2. Makes the .ssh directory if it doesn't already exist to avoid crashing
3. Wraps the identity file path in quotes in case of spaces in the filename

Happy to rework anything as required

### CODE REVIEW

File: src/Documenter.jl [hunk 1]
```diff
@@ -599,15 +599,15 @@ and when building docs for a tag they are deployed to a `vX.Y.Z` directory.
 function git_push(
         root, temp, repo;
         branch="gh-pages", dirname="", target="site", tag="", key="", sha="", devurl="dev",
-        versions, forcepush=false,
+        versions, forcepush=false, host="github.com",
```
Comment (fredrikekre) [inline src/Documenter.jl]: Are you planning to call `git_push` directly or where is this kwarg set? Couldn't we find the host from the `repo` argument to `deploydocs`?

File: src/Documenter.jl [hunk 2]
```diff
@@ -624,12 +624,14 @@ function git_push(
 
     try
         # Use a custom SSH config file to avoid overwriting the default user config.
+        ssh_dir = joinpath(homedir(), ".ssh")
+        isdir(ssh_dir) || mkdir(ssh_dir)
```
Comment (fredrikekre) [inline src/Documenter.jl]: This should be cleaned up if it was created.

Comment (JackDunnNZ) [reply src/Documenter.jl]: I moved this logic into `withfile` to create and clean up the directory

File: src/Documenter.jl (hunk 1 above)
Comment (JackDunnNZ) [reply src/Documenter.jl]: Yes currently I call `git_push` directly as in the first comment. We could definitely pull the host from the `repo` argument as something like `split(repo, "/")[1]` if you think that would be better - it's probably more robust.

Comment (mortenpi) [reply src/Documenter.jl]: Yeah, parsing `repo` seems better here.

I think currently we allow both `github.com:` and `github.com/` as the start of the `repo`. So we could formalize that a bit and say repo has to start with the domain name, then you have `/` or `:` as a separator, and then the rest of the URL follows. So we should also split by `:`, or probably use a regex to match the start.

**Comment by mortenpi** [comment]
Other than the `repo`/`host` thing, LGTM! @JackDunnNZ could you also drop a small note into `CHANGELOG.md`?

**Comment by JackDunnNZ** [comment]
Thanks both for the comments! I've changed to extract the host using regex and added the note to the changelog

**Comment by mortenpi** [comment]
Thanks!

**Pull request merged**